src/routes/signin/pat.ts

```diff
@@ -0,0 +1,37 @@
+import { sendError } from '@/errors';
+import { ENV, createHasuraAccessToken, getUserByRefreshToken } from '@/utils';
+import { personalAccessToken } from '@/validation';
+import { RequestHandler } from 'express';
+import Joi from 'joi';
+
+export const signInPATSchema = Joi.object({
+  personalAccessToken,
+}).meta({ className: 'SignInPATSchema' });
+
+export const signInPATHandler: RequestHandler<
+  {},
+  {},
+  { personalAccessToken: string }
+> = async (req, res) => {
+  const user = await getUserByRefreshToken(req.body.personalAccessToken);
```

Comment:
should we also validate the type of the refresh token is `PAT`?

Reply:
Yeah, good point. I'll create a separate GraphQL query for this one.